### PR TITLE
[REVIEW] Add support for parquet INT96 timestamps

### LINKED ISSUE #1499: [BUG] Read_parquet not working with spark timestamp type in parquet file
**Describe the bug**
Reading a parquet file results in error.

**Steps/Code to reproduce bug**
```
import cudf
cudf.io.read_parquet('/path/to/file.snappy.parquet')
```
Output:
```
RuntimeError                              Traceback (most recent call last)
/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+834.g3985c79.dirty-py3.7-linux-x86_64.egg/cudf/io/parquet.py in read_parquet(path, engine, columns, *args, **kwargs)
     17         df = cpp_read_parquet(
     18             path,
---> 19             columns
     20         )
     21     else:

cudf/bindings/parquet.pyx in cudf.bindings.parquet.cpp_read_parquet()

cudf/bindings/parquet.pyx in cudf.bindings.parquet.cpp_read_parquet()

RuntimeError: Invalid gdf_dtype in type_dispatcher
```

Also when trying to read a non existent file:
```
import cudf
cudf.io.read_parquet('/path/to/nonexistent/file')
```
Output
```
/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+834.g3985c79.dirty-py3.7-linux-x86_64.egg/cudf/io/parquet.py in read_parquet(path, engine, columns, *args, **kwargs)
     17         df = cpp_read_parquet(
     18             path,
---> 19             columns
     20         )
     21     else:

cudf/bindings/parquet.pyx in cudf.bindings.parquet.cpp_read_parquet()

cudf/bindings/parquet.pyx in cudf.bindings.parquet.cpp_read_parquet()

NameError: name 'errno' is not defined
```
**Expected behavior**
File gets read if valid. If filename is invalid, the proper error gets propagated. 
**Environment details (please complete the following information):**
 - Method of cuDF install: From source

### PR DESCRIPTION
**Pull request by OlivierNV**
Convert parquet INT96 (Spark timestamps) to GDF_DATE64[ms]
Add a corresponding test parquet file under data/parquet/

Resolves #1499

### CODE REVIEW

File: cpp/src/io/parquet/page_data.cu [hunk 1]
```diff
@@ -763,6 +763,66 @@ inline __device__ void gpuStoreOutput(uint2 *dst, const uint8_t *src8, uint32_t
 }
 
 
+/**
+ * @brief Convert an INT96 Spark timestamp to 64-bit timestamp
+ *
+ * @param[out] dst ptr to output
+ * @param[in] src8 raw input bytes
+ * @param[in] dict_pos byte position in dictionary
+ * @param[in] dict_size size of dictionary
```
Comment (j-ieong) [inline cpp/src/io/parquet/page_data.cu]: Outdated doxygen params for the function?

Comment (OlivierNV) [reply cpp/src/io/parquet/page_data.cu]: Thanks, updated.

File: cpp/src/io/parquet/parquet_gpu.h [hunk 2]
```diff
@@ -22,6 +22,8 @@
 
 namespace parquet { namespace gpu {
 
+#define INT96_TS_CLKRATE        1000 // Clock frequency for INT96 timestamp conversion (1000=ms, 1000000000=ns)
```
[the remark this answers is not in the document]
Comment (OlivierNV) [reply cpp/src/io/parquet/parquet_gpu.h]: I prefer the define, but I suppose I'm too oldschool, so willing to go with constexpr: is there an equivalent to #if (INT96_TS_CLKRATE == 1000) in the pre-processing stage with constrexpr, or do I have to rely on compiler optimizing out the constrexpr statement ?

[the remark this answers is not in the document]
Comment (vuule) [reply cpp/src/io/parquet/parquet_gpu.h]: constexpr is guaranteed to be evaluated the compile time, so it should work everywhere macros work. I'm not sure about that specific use, but it should be fine.

**Comment by j-ieong** [comment]
> Convert parquet INT96 (Spark timestamps) to GDF_DATE64[ms]
> Add a corresponding test parquet file under data/parquet/
> 
> Resolves issue #1499

You'll want to edit the message to remove the `issue` word and just say "Resolves #1499".

File: cpp/src/io/parquet/parquet_gpu.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (OlivierNV) [reply cpp/src/io/parquet/parquet_gpu.h]: It does seem that the benefits of constexpr here would be purely of the cosmetic kind :)

**Comment by OlivierNV** [comment]
> You'll want to edit the message to remove the `issue` word and just say "Resolves #1499". The issue currently only says this PR references it and not closes it.

Done, thanks!

**Comment by mjsamoht** [comment]
@vuule I believe Olivier addressed your suggestion.  Let me know if you're good with this PR now.

**Pull request merged**